### PR TITLE
QL: Adapt nested fields extraction from "fields" API output to the new un-flattened strucutre

### PR DESCRIPTION
**Pull request by astefan**
"fields" API ouput for nested fields is returning the values in a way that keeps the relationship between nested documents
Addresses https://github.com/elastic/elasticsearch/issues/68722

### CODE REVIEW

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java [hunk 1]
```diff
@@ -78,7 +79,13 @@ public void writeTo(StreamOutput out) throws IOException {
     @Override
     public Object extract(SearchHit hit) {
         Object value = null;
-        DocumentField field = hit.field(fieldName);
+        DocumentField field = null;
+        if (hitName != null) {
+            // a nested field value is grouped under the nested parent name (ie dep.dep_name lives under "dep":[{dep_name:value}])
```
Comment (bpintea) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java:84]:
```suggestion
            // a nested field value is grouped under the nested parent name (ie dep.dep_name lives under "dep":[{"dep_name":value}])
```

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java [hunk 2]
```diff
@@ -78,7 +79,13 @@ public void writeTo(StreamOutput out) throws IOException {
...
+            field = hit.field(hitName);
+        } else {
+            field = hit.field(fieldName);
```
Comment (bpintea) [inline x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java:87]: To make sure, the fieldName is in this case a "doc-top-level" field, right?

**Review by bpintea — APPROVED** [review]
LGTM

File: x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java (hunk 2 above)
Comment (astefan) [reply x-pack/plugin/ql/src/main/java/org/elasticsearch/xpack/ql/execution/search/extractor/AbstractFieldHitExtractor.java:87]: `fieldName` is the full field name (including the path): `dep.dep_name` for example.

**Review by matriv — APPROVED** [review]
LGTM

**Review by costin — APPROVED** [review]
LGTM. Thanks for the fast turnaround.

**Review by palesz — APPROVED** [review]
LGTM

**Pull request merged**